--- freeradius-1.1.2/raddb/radiusd.conf.in.no_sql_inc	2006-08-15 11:38:23.000000000 +0200
+++ freeradius-1.1.2/raddb/radiusd.conf.in	2006-08-15 11:38:25.000000000 +0200
@@ -1245,7 +1245,7 @@
 	#  For MS-SQL, use:	 	${confdir}/mssql.conf
 	#  For Oracle, use:	 	${confdir}/oraclesql.conf
 	#
-	$INCLUDE  ${confdir}/sql.conf
+#	$INCLUDE  ${confdir}/sql.conf
 
 
 	#  For Cisco VoIP specific accounting with Postgresql,
