diff -up freeradius-1.1.7/src/modules/rlm_ldap/rlm_ldap.c.ldap freeradius-1.1.7/src/modules/rlm_ldap/rlm_ldap.c
--- freeradius-1.1.7/src/modules/rlm_ldap/rlm_ldap.c.ldap	2007-06-18 11:00:27.000000000 +0200
+++ freeradius-1.1.7/src/modules/rlm_ldap/rlm_ldap.c	2007-08-06 14:40:31.000000000 +0200
@@ -1041,7 +1041,7 @@ static int ldap_groupcmp(void *instance,
 		radlog(L_ERR, "rlm_ldap: Add ldap connections are in use");
 		return 1;
 	}
-	if ((res = perform_search(inst, conn, vp_user_dn->strvalue, LDAP_SCOPE_BASE,
+	if ((res = perform_search(inst, conn, (char *)vp_user_dn->strvalue, LDAP_SCOPE_BASE,
 					filter, group_attrs,&result)) != RLM_MODULE_OK){
 		DEBUG("rlm_ldap::ldap_groupcmp: Search returned error");
 		ldap_release_conn(conn_id, inst->conns);
@@ -1356,7 +1356,7 @@ ldap_authorize(void *instance, REQUEST *
 
 		strNcpy(filter,inst->base_filter,sizeof(filter));
 		if (user_profile)
-			profile = user_profile->strvalue;
+			profile = (char *)user_profile->strvalue;
 		if (profile && strlen(profile)){
 			if ((res = perform_search(instance, conn,
 				profile, LDAP_SCOPE_BASE,
@@ -1460,7 +1460,7 @@ ldap_authorize(void *instance, REQUEST *
 							}
 							added_known_password = 1;
 							passwd_len = strlen(passwd_val);
-							strncpy(passwd_item->strvalue,passwd_val,MAX_STRING_LEN - 1);
+							strncpy((char *)passwd_item->strvalue,passwd_val,MAX_STRING_LEN - 1);
 							passwd_item->length = (passwd_len > (MAX_STRING_LEN - 1)) ? (MAX_STRING_LEN - 1) : passwd_len;
 							pairadd(&request->config_items,passwd_item);
 							DEBUG("rlm_ldap: Added password %s in check items",passwd_item->strvalue);
@@ -1804,12 +1804,12 @@ ldap_authenticate(void *instance, REQUES
 		ldap_msgfree(result);
 	}
 
-	user_dn = vp_user_dn->strvalue;
+	user_dn = (char *)vp_user_dn->strvalue;
 
 	DEBUG("rlm_ldap: user DN: %s", user_dn);
 
 #ifndef NOVELL
-	ld_user = ldap_connect(instance, user_dn, request->password->strvalue,
+	ld_user = ldap_connect(instance, user_dn, (char *)request->password->strvalue,
 			       1, &res, NULL);
 #else
 
@@ -1954,7 +1954,7 @@ retry:
 		}
 	}
 
-	ld_user = ldap_connect(instance, user_dn, request->password->strvalue,
+	ld_user = ldap_connect(instance, user_dn, (char *)request->password->strvalue,
 			1, &res, &err);
 
 	if(err != NULL){
@@ -2208,7 +2208,7 @@ ldap_connect(void *instance, const char 
 
 #ifdef HAVE_LDAP_INT_TLS_CONFIG
 
-	if ( ldap_int_tls_config( NULL, LDAP_OPT_X_TLS_REQUIRE_CERT,
+	if ( ldap_set_option( NULL, LDAP_OPT_X_TLS_REQUIRE_CERT,
 							  (inst->tls_require_cert) )
 		 != LDAP_OPT_SUCCESS) {
 		radlog(L_ERR, "rlm_ldap: could not set "
@@ -2622,7 +2622,7 @@ ldap_pairget(LDAP * ld, LDAPMessage * en
 
 				if (do_xlat) {
 					newpair->flags.do_xlat = 1;
-					strNcpy(newpair->strvalue, buf,
+					strNcpy((char *)newpair->strvalue, buf,
 						sizeof(newpair->strvalue));
 					newpair->length = 0;
 				}
