--- freeradius-1.0.0-pre3/src/modules/rlm_eap/eap.c.ltdl_no_la	2004-05-05 19:20:40.000000000 +0300
+++ freeradius-1.0.0-pre3/src/modules/rlm_eap/eap.c	2004-06-29 12:33:04.000000000 +0300
@@ -82,7 +82,11 @@
 	snprintf(buffer, sizeof(buffer), "rlm_eap_%s", eaptype_name);
 
 	/* Link the loaded EAP-Type */
-	handle = lt_dlopenext(buffer);
+	char *tmp = malloc(strlen(buffer) + 4);
+	strcpy(tmp, buffer);
+	strcat(tmp, ".so");
+	handle = lt_dlopenext(tmp);
+	free(tmp);
 	if (handle == NULL) {
 		radlog(L_ERR, "rlm_eap: Failed to link EAP-Type/%s: %s",
 		       eaptype_name, lt_dlerror());
--- freeradius-1.0.0-pre3/src/main/modules.c.ltdl_no_la	2004-04-23 22:50:29.000000000 +0300
+++ freeradius-1.0.0-pre3/src/main/modules.c	2004-06-29 12:30:42.000000000 +0300
@@ -204,7 +204,11 @@
 	/*
 	 *	Keep the handle around so we can dlclose() it.
 	 */
-	handle = lt_dlopenext(module_name);
+	char *tmp = malloc(strlen(module_name) + 4);
+	strcpy(tmp, module_name);
+	strcat(tmp, ".so");
+	handle = lt_dlopenext(tmp);
+	free(tmp);
 	if (handle == NULL) {
 		radlog(L_ERR|L_CONS, "%s[%d] Failed to link to module '%s':"
 				" %s\n", cffilename, cflineno, module_name, lt_dlerror());
--- freeradius-1.0.0-pre3/src/modules/rlm_sql/rlm_sql.c.ltdl_no_la	2004-05-25 09:56:43.000000000 +0300
+++ freeradius-1.0.0-pre3/src/modules/rlm_sql/rlm_sql.c	2004-06-29 12:30:42.000000000 +0300
@@ -668,7 +668,11 @@
 		return -1;
 	}
 
-	inst->handle = lt_dlopenext(inst->config->sql_driver);
+	char *tmp = malloc(strlen(inst->config->sql_driver) + 4);
+	strcpy(tmp, inst->config->sql_driver);
+	strcat(tmp, ".so");
+	inst->handle = lt_dlopenext(tmp);
+	free(tmp);
 	if (inst->handle == NULL) {
 		radlog(L_ERR, "rlm_sql (%s): Could not link driver %s: %s",
 		       inst->config->xlat_name, inst->config->sql_driver,
