--- freeradius-server-2.0.5/raddb/radiusd.conf.in.conf	2008-05-30 05:18:43.000000000 -0400
+++ freeradius-server-2.0.5/raddb/radiusd.conf.in	2008-07-30 12:44:54.000000000 -0400
@@ -627,7 +627,7 @@
 	#  Include another file that has the SQL-related configuration.
 	#  This is another file only because it tends to be big.
 	#
-	$INCLUDE sql.conf
+	#$INCLUDE sql.conf
 
 
 	#  For Cisco VoIP specific accounting with Postgresql,
@@ -648,7 +648,7 @@
 	#  totally dependent on the SQL module to process Accounting
 	#  packets.
 	#
-	$INCLUDE sql/mysql/counter.conf
+	#$INCLUDE sql/mysql/counter.conf
 	#$INCLUDE sql/postgresql/counter.conf
 
 	# $INCLUDE sqlippool.conf
